### ISSUE TITLE
无法输出验证集的loss

### ISSUE THREAD

**Issue by Logosxxw** [hiyouga/ChatGLM-Efficient-Tuning#93]
除了BLEU和Rouge之外，也想看模型在验证集上的loss，但阅读源码之后发现验证集的loss直接被置为None，导致无法输出验证集的loss。

src/utils/seq2seq.py的210行
Seq2SeqTrainerForChatGLM类的prediction_step方法，由于计算BLEU和Rouge不需要依赖loss，代码直接设置了：`loss = None`

希望能够实现loss的计算逻辑，方便do_eval的时候输出验证集loss，感谢

**Comment by GUORUIWANG** [comment]
+1期待

**Comment by hiyouga** [comment]
在验证时不加入 `--prediction_with_generate` 参数即可输出验证集的 loss。例如：
```bash
CUDA_VISIBLE_DEVICES=0 python src/train_sft.py \
    --do_eval \
    --dataset alpaca_gpt4_zh \
    --checkpoint_dir path_to_checkpoint \
    --output_dir path_to_eval_result \
    --overwrite_cache \
    --per_device_eval_batch_size 8 \
    --max_samples 100
```
